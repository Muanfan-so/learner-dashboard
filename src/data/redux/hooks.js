import { useSelector } from 'react-redux';

import { actions as appActions } from './app/reducer';
import appSelectors from './app/selectors';
import requestSelectors from './requests/selectors';

const { courseCard } = appSelectors;

export const usePageNumber = () => useSelector(appSelectors.pageNumber);
export const useEmailConfirmationData = () => useSelector(appSelectors.emailConfirmation);
export const useEnterpriseDashboardData = () => useSelector(appSelectors.enterpriseDashboard);
export const usePlatformSettingsData = () => useSelector(appSelectors.platformSettings);
// suggested courses is max at 3 at the moment.
export const useSuggestedCoursesData = () => useSelector(appSelectors.suggestedCourses).slice(0, 3);
export const useRecommendedCoursesData = () => useSelector(appSelectors.recommendedCourses);
export const useSelectSessionModalData = () => useSelector(appSelectors.selectSessionModal);

export const useHasCourses = () => useSelector(appSelectors.hasCourses);
export const useHasAvailableDashboards = () => useSelector(appSelectors.hasAvailableDashboards);
export const useCurrentCourseList = (opts) => useSelector(
  state => appSelectors.currentList(state, opts),
);
export const useShowSelectSessionModal = () => useSelector(appSelectors.showSelectSessionModal);

// eslint-disable-next-line
export const useCourseCardData = (selector) => (cardId) => useSelector(
  (state) => selector(state, cardId),
);

export const useCardCertificateData = useCourseCardData(courseCard.certificate);
export const useCardCourseData = useCourseCardData(courseCard.course);
export const useCardCourseRunData = useCourseCardData(courseCard.courseRun);
export const useCardEnrollmentData = useCourseCardData(courseCard.enrollment);
export const useCardEntitlementData = useCourseCardData(courseCard.entitlement);
export const useCardGradeData = useCourseCardData(courseCard.gradeData);
export const useCardProviderData = useCourseCardData(courseCard.courseProvider);
export const useCardRelatedProgramsData = useCourseCardData(courseCard.relatedPrograms);

export const useUpdateSelectSessionModalCallback = (dispatch, cardId) => () => dispatch(
  appActions.updateSelectSessionModal(cardId),
);

export const useMasqueradeData = () => useSelector(requestSelectors.masquerade);

export const useIsPendingRequest = (requestName) => useSelector(requestSelectors.isPending(requestName));
