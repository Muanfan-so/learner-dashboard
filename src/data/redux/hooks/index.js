import * as app from './app';
import * as requests from './requests';

export * from './app';
export { requests };

export default { app, requests };
