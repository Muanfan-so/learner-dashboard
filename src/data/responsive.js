import { useWindowSize, breakpoints } from '@edx/paragon';
import { StrictDict } from 'utils';

const createResponsive = (fn) => () => {
  const { width } = useWindowSize();
  return fn(width);
};

export const isDesktopSize = createResponsive(
  (width) => width >= breakpoints.large.minWidth,
);

// Beware of using isPadSize() for styling. When the screen resizes pass the
// breakpoint, the screen will flash between the two styles. This is why it is
// recommened isDesktopSize() or isMobileSize() be used instead.
export const isIpadSize = createResponsive(
  (width) => width >= breakpoints.medium.minWidth && width < breakpoints.large.minWidth,
);

export const isMobileSize = createResponsive(
  (width) => width < breakpoints.medium.minWidth,
);

export default StrictDict({
  isDesktopSize,
  isIpadSize,
  isMobileSize,
});
