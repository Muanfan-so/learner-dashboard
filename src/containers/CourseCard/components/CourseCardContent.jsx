import React from 'react';
import PropTypes from 'prop-types';
import { useIntl } from '@edx/frontend-platform/i18n';

// import PropTypes from 'prop-types';
import { Card } from '@edx/paragon';

import { hooks as appHooks } from 'data/redux';

import RelatedProgramsBadge from './RelatedProgramsBadge';
import CourseCardMenu from './CourseCardMenu';
import CourseCardActions from './CourseCardActions';
import CourseCardDetails from './CourseCardDetails';

import messages from '../messages';

export const CourseCardContent = ({ cardId, orientation }) => {
  const { formatMessage } = useIntl();
  const { courseName, bannerImgSrc } = appHooks.useCardCourseData(cardId);
  const { homeUrl } = appHooks.useCardCourseRunData(cardId);
  return (
    <>
      <a className="pgn__card-wrapper-image-cap horizontal" href={homeUrl}>
        <img
          className="pgn__card-image-cap"
          src={bannerImgSrc}
          alt={formatMessage(messages.bannerAlt)}
        />
      </a>
      <Card.Body>
        <Card.Header
          title={(
            <a href={homeUrl} data-testid="CourseCardTitle">
              {courseName}
            </a>
          )}
          actions={<CourseCardMenu cardId={cardId} />}
        />
        <Card.Section className="pt-0">
          <CourseCardDetails cardId={cardId} />
        </Card.Section>
        {orientation === 'vertical'
          ? (
            <>
              <RelatedProgramsBadge cardId={cardId} />
              <Card.Footer orientation="horizontal">
                <CourseCardActions cardId={cardId} />
              </Card.Footer>
            </>
          ) : (
            <Card.Footer
              orientation="vertical"
              textElement={<RelatedProgramsBadge cardId={cardId} />}
            >
              <CourseCardActions cardId={cardId} />
            </Card.Footer>
          )}
      </Card.Body>
    </>
  );
};

CourseCardContent.propTypes = {
  cardId: PropTypes.string.isRequired,
  orientation: PropTypes.string.isRequired,
};

CourseCardContent.defaultProps = {};

export default CourseCardContent;
