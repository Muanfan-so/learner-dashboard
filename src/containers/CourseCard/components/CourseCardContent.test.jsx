import { shallow } from 'enzyme';

import { hooks } from 'data/redux';
import CourseCardContent from './CourseCardContent';

jest.mock('data/redux', () => ({
  hooks: {
    useCardCourseData: jest.fn(),
  },
}));

jest.mock('./CourseCardActions', () => 'CourseCardActions');
jest.mock('./CourseCardDetails', () => 'CourseCardDetails');
jest.mock('./RelatedProgramsBadge', () => 'RelatedProgramsBadge');
jest.mock('./CourseCardMenu', () => 'CourseCardMenu');

describe('CourseCardContent', () => {
  const props = {
    cardId: 'test-card-id',
    orientation: 'vertical',
  };
  hooks.useCardCourseData.mockReturnValue({
    courseName: 'test-course-name',
    bannerImgSrc: 'test-banner-img-src',
  });
  describe('snapshot', () => {
    test('orientation vertical', () => {
      const wrapper = shallow(<CourseCardContent {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
    test('orientation horizontal', () => {
      const wrapper = shallow(<CourseCardContent {...props} orientation="horizontal" />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
