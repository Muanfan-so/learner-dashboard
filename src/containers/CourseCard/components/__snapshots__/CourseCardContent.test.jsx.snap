// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCardContent snapshot not verified 1`] = `
<Fragment>
  <a
    className="pgn__card-wrapper-image-cap overflow-visible vertical"
    href="test-home-url"
  >
    <img
      alt="Course thumbnail"
      className="pgn__card-image-cap"
      src="test-banner-img-src"
    />
  </a>
  <Card.Body>
    <Card.Header
      actions={
        <CourseCardMenu
          cardId="test-card-id"
        />
      }
      title={
        <h3>
          <a
            className="course-card-title"
            data-testid="CourseCardTitle"
            href="test-home-url"
          >
            test-course-name
          </a>
        </h3>
      }
    />
    <Card.Section
      className="pt-0"
    >
      <CourseCardDetails
        cardId="test-card-id"
      />
    </Card.Section>
    <RelatedProgramsBadge
      cardId="test-card-id"
    />
    <Card.Footer
      orientation="horizontal"
    >
      <CourseCardActions
        cardId="test-card-id"
      />
    </Card.Footer>
  </Card.Body>
</Fragment>
`;

exports[`CourseCardContent snapshot orientation horizontal 1`] = `
<Fragment>
  <a
    className="pgn__card-wrapper-image-cap overflow-visible horizontal"
    href="test-home-url"
  >
    <img
      alt="Course thumbnail"
      className="pgn__card-image-cap"
      src="test-banner-img-src"
    />
    <span
      className="course-card-verify-ribbon-container"
      title="You're enrolled as a verified student"
    >
      <Badge
        as="div"
        className="w-100"
        variant="success"
      >
        Verified
      </Badge>
      <img
        alt="ID Verified Ribbon/Badge"
        src="test-file-stub"
      />
    </span>
  </a>
  <Card.Body>
    <Card.Header
      actions={
        <CourseCardMenu
          cardId="test-card-id"
        />
      }
      title={
        <h3>
          <a
            className="course-card-title"
            data-testid="CourseCardTitle"
            href="test-home-url"
          >
            test-course-name
          </a>
        </h3>
      }
    />
    <Card.Section
      className="pt-0"
    >
      <CourseCardDetails
        cardId="test-card-id"
      />
    </Card.Section>
    <Card.Footer
      orientation="vertical"
      textElement={
        <RelatedProgramsBadge
          cardId="test-card-id"
        />
      }
    >
      <CourseCardActions
        cardId="test-card-id"
      />
    </Card.Footer>
  </Card.Body>
</Fragment>
`;

exports[`CourseCardContent snapshot orientation vertical 1`] = `
<Fragment>
  <a
    className="pgn__card-wrapper-image-cap overflow-visible vertical"
    href="test-home-url"
  >
    <img
      alt="Course thumbnail"
      className="pgn__card-image-cap"
      src="test-banner-img-src"
    />
    <span
      className="course-card-verify-ribbon-container"
      title="You're enrolled as a verified student"
    >
      <Badge
        as="div"
        className="w-100"
        variant="success"
      >
        Verified
      </Badge>
      <img
        alt="ID Verified Ribbon/Badge"
        src="test-file-stub"
      />
    </span>
  </a>
  <Card.Body>
    <Card.Header
      actions={
        <CourseCardMenu
          cardId="test-card-id"
        />
      }
      title={
        <h3>
          <a
            className="course-card-title"
            data-testid="CourseCardTitle"
            href="test-home-url"
          >
            test-course-name
          </a>
        </h3>
      }
    />
    <Card.Section
      className="pt-0"
    >
      <CourseCardDetails
        cardId="test-card-id"
      />
    </Card.Section>
    <RelatedProgramsBadge
      cardId="test-card-id"
    />
    <Card.Footer
      orientation="horizontal"
    >
      <CourseCardActions
        cardId="test-card-id"
      />
    </Card.Footer>
  </Card.Body>
</Fragment>
`;
